grails-core/src/main/groovy/org/grails/core/cfg/GroovyConfigPropertySourceLoader.groovy

```diff
@@ -50,7 +50,6 @@ class GroovyConfigPropertySourceLoader implements PropertySourceLoader {
     }
 
     List<PropertySource<?>> load(String name, Resource resource, List<String> filteredKeys) throws IOException {
-        if (loaded.compareAndSet(false, true)) {
```

Comment:
Question for @graemerocher 

I removed this check. With this done, on the example app you can place `application.groovy` and `application-development.groovy`, Or `application.yml` and `application-development.groovy` and the correct properties are read each time. You can also mix and match the `application-test.yml` with `application.groovy` or whatever you like and it is correctly reading the properties.

However, I am not sure if removing the `loaded` variable is the correct this to do.  Removing it does not break any tests. So I looked at the history to see if I can speculate why it was placed there.

The best I can come up with is that it was carried over in the spirit of  `ConfigSlurper`, but now its no longer relevant. And that file did not need to be read twice. That is because all the environment declarations were in one file. Now with this approach, we have groovy files for every environment. Therefore I though to simply remove it

Please look over my assumptions. I derived the thought above from looking at the history of `GroovyConfigPropertySourceLoader` and saw this checkin:

https://github.com/grails/grails-core/commit/1b79e87681b63e16ead3bd76e6ce4b140a05039b#diff-3726c3b36d86fb338209f4451c4b50f3L50

Reply:
"all the environment declarations were in one file" 

That is still a supported approach so we need to ensure the same file isn't read twice

Reply:
ok i will attempt to keep track of the file names that are read, instead of one general boolean flag.